examples/perplexity/perplexity.cpp

```diff
@@ -112,6 +112,43 @@ static results_log_softmax log_softmax(int n_vocab, const float * logits, int to
     return {logits[tok] - max_logit - log(sum_exp), logits[tok], expf(logits[tok] - max_logit) / (float) sum_exp};
 }
 
+static inline int nearest_int(float fval) {
+    //assert(fval <= 4194303.f);
+    float val = fval + 12582912.f;
+    int i; memcpy(&i, &val, sizeof(int));
+    return (i & 0x007fffff) - 0x00400000;
+}
```

Comment:
This function has been in `ggml` forever (see https://github.com/ggerganov/llama.cpp/blob/6f9939d119b2d004c264952eb510bd106455531e/ggml-quants.c#L1241). As it is not exposed, I decided to copy/paste it here. It is many times faster than `round()` or similar. Which is understandable because the standard library functions need to deal with all sort of cases. Here we have a known range (0...65535) for the floats we are rounding, so we can take advantage of that. One could also simply have used `int(fval + 0.5f)`, except for that unfortunate max value that may round to 65536 that way, which then requires a separate check.